charms/mlflow-server/src/charm.py

```diff
@@ -276,6 +277,22 @@ def _check_image_details(self):
         return image_details
 
 
+def validate_s3_bucket_name(name):
+    """Validates the name as a valid S3 bucket name, raising a CheckFailedError if invalid."""
+    # regex from https://stackoverflow.com/a/50484916/5394584
+    if re.match(
+        r"(?=^.{3,63}$)(?!^(\d+\.)+\d+$)(^(([a-z0-9]|[a-z0-9][a-z0-9\-]*[a-z0-9])\.)*([a-z0-9]|[a-z0-9][a-z0-9\-]*[a-z0-9])$)",
```

Comment:
I'm not a big fan of `re`, but since we are doing this, we could use `re.compile` here as we are going to be calling this many times during the charm's lifecycle.